src/Database/Migrations/2022-12-13-200557_add_ban_columns.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Database\Migrations;
+
+use CodeIgniter\Database\Forge;
+use CodeIgniter\Database\Migration;
+use CodeIgniter\Shield\Config\Auth;
+
+class AddBanColumns extends Migration
+{
+    /**
+     * Auth Table names
+     */
+    private array $tables;
+
+    public function __construct(?Forge $forge = null)
+    {
+        parent::__construct($forge);
+
+        /** @var Auth $authConfig */
+        $authConfig   = config('Auth');
+        $this->tables = $authConfig->tables;
+    }
+
+    public function up(): void
+    {
+        // Users Table
+        $fields = [
+            'banned'         => ['type' => 'tinyint', 'after' => 'active', 'constraint' => 1, 'null' => false, 'default' => 0],
+            'banned_message' => ['type' => 'varchar', 'after' => 'banned', 'constraint' => 255, 'null' => true],
```

Comment:
If `banned` and `banned_message` is going to be used instead of `status` and `status_message`, it is better, rename the previous field (`$forge->renameTable`) and use command (`$forge->modifyColumn()`) to change the properties.

This may seem like a **breaking change**, but I don't think it will bother many users. Because we have not been able to use `status` and `status_message` before.

This is my opinion, see what others say.

Reply:
I was thinking the status can be utilised in the future for setting the status of a user. For example, a user may say set his status to away and then have a message to add context to the status that they set.

Reply:
It seems Myth::Auth uses `status` for banning.
https://github.com/lonnieezell/myth-auth/blob/1c646c58e8b9b956b2163ebda8e5ec7e9ed609ce/src/Entities/User.php#L196-L202

Basically, Shield should use the minimum number of columns required.
We are hesitant to use the `status` because the use of the `status` is not clear.
If the `status` is the status of the user for administration, ban should use the `status`.

Reply:
@datamweb Renaming the column is a breaking change. If a dev already uses the `status` for some reason, the app will break.

But if a dev uses the `status` for user's current status (away, busy, etc.), then the app will break even if we use the `status` for banning.

After all, it is not good to have columns with unclear uses that are not being used.

Reply:
@kenjis @davidnsai I would like to use `is_banned` rather than `banned` column with `tinyint` type. 

`banned_message` is for me extra column which is not need I rather to use existing `status` column for message. 

If user is banned then he just need to get message within login that he can't login because he was banned and his potentially set `status` like away etc. needs to be cleared after all.

Reply:
I thought about this.

Shield is an Authentication and Authorization library.

If the `status` is for user's current status (away, busy, etc.), it is not for Authentication,
nor for Authorization. We should not provide the `status`.

Essentially, banning is about authorization. So it can be implemented using the authorization function. 
However, authorization does not have the ability to provide (have) a message for a user in case of non-authorization.

It seems no problem that we use the `status` for the user status for users who can be authenticated but cannot login now in some reason. In this case, we can use the `status` for banning, account locking with too many login failures.

Reply:
I think you are right there @kenjis. This means that the 2 new columns introduced for `banned` and `banned_message` are redundant but it now allows us to use the `status` field for a multitude of features without having to define a new column for every new feature that may be added in future. So in this case the `status` field would contain `banned` in a different scenario it may contain `locked` etc.

Reply:
@lonnieezell I just need a resolution on whether I should get rid of the banned and banned_message fields and use the status and status_message fields then the feature should be ready.

Reply:
@lonnieezell See this thread.
I hope this PR uses the `status` column for banning like Myth::Auth.

Reply:
I had read this last night. I thought it was resolved to use the `status` column, which I agree with. That was the original intention for that column - to be used for a variety of system-related statuses, like banning, suspension, etc.